Description: PMIX_MODEX removed in latest PMIX releases (>= 3.1.0 rc2)
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2019-01-01
Forwarded: no

Index: openmpi-3.1.3/opal/mca/pmix/pmix2x/pmix2x.c
===================================================================
--- openmpi-3.1.3.orig/opal/mca/pmix/pmix2x/pmix2x.c
+++ openmpi-3.1.3/opal/mca/pmix/pmix2x/pmix2x.c
@@ -1001,10 +1001,6 @@ int pmix2x_value_unload(opal_value_t *kv
         OPAL_ERROR_LOG(OPAL_ERR_NOT_SUPPORTED);
         rc = OPAL_ERR_NOT_SUPPORTED;
         break;
-    case PMIX_MODEX:
-        OPAL_ERROR_LOG(OPAL_ERR_NOT_SUPPORTED);
-        rc = OPAL_ERR_NOT_SUPPORTED;
-        break;
     case PMIX_PERSIST:
         kv->type = OPAL_PERSIST;
         kv->data.uint8 = pmix2x_convert_persist(v->data.persist);
@@ -1111,10 +1107,6 @@ int pmix2x_value_unload(opal_value_t *kv
         OPAL_ERROR_LOG(OPAL_ERR_NOT_SUPPORTED);
         rc = OPAL_ERR_NOT_SUPPORTED;
         break;
-    case PMIX_INFO_ARRAY:
-        OPAL_ERROR_LOG(OPAL_ERR_NOT_SUPPORTED);
-        rc = OPAL_ERR_NOT_SUPPORTED;
-        break;
 
     default:
         /* silence warnings */
Index: openmpi-3.1.3/opal/mca/pmix/ext1x/pmix1x_server_south.c
===================================================================
--- openmpi-3.1.3.orig/opal/mca/pmix/ext1x/pmix1x_server_south.c
+++ openmpi-3.1.3/opal/mca/pmix/ext1x/pmix1x_server_south.c
@@ -41,6 +41,11 @@
 #include "pmix.h"
 #include "pmix_server.h"
 
+/* Deprecated. Rewrite code below */
+#ifndef PMIX_INFO_ARRAY
+#define PMIX_INFO_ARRAY 44
+#endif
+
 /****    S.O.U.T.H.B.O.U.N.D   I.N.T.E.R.F.A.C.E.S     ****/
 
 /* These are the interfaces used by the OMPI/ORTE/OPAL layer to call
